src/ERC20Bridgeable/libraries/LibERC20Bridgeable.sol

```diff
@@ -0,0 +1,145 @@
+        uint256 totalSupply;
+        mapping(address owner => uint256 balance) balanceOf;
+    }
+
+    /**
+     * @notice Returns the ERC20 storage struct from the predefined diamond storage slot.
+     * @dev Uses inline assembly to set the storage slot reference.
+     * @return s The ERC20 storage struct reference.
+     */
+    function erc20Storage() internal pure returns (ERC20Storage storage s) {
+        bytes32 position = ERC20_STORAGE_POSITION;
+        assembly {
+            s.slot := position
+        }
+    }
+    /// @notice Storage slot for ERC-20 Bridgeable using ERC8042 as template
+
+    bytes32 constant BRIDGEABLE_STORAGE_POSITION = keccak256("compose.erc20.bridgeable");
+
+    struct ERC20BridgeableStorage {
+        mapping(address => bool) trustedBridges;
+    }
+
+    function bridgeStorage() internal pure returns (ERC20BridgeableStorage storage s) {
+        bytes32 position = BRIDGEABLE_STORAGE_POSITION;
+
+        assembly {
+            s.slot := position
+        }
+    }
+
+    // the bridge account must be a trusted account
+    /// @notice Internal crosschain mint logic. MUST be called only after validating caller.
+    /// @dev Increases totalSupply and recipient balance and emits CrosschainMint.
+    /// @param _account The account to mint tokens to.
+    /// @param _value The amount to mint.
+    function crosschainMint(address _account, uint256 _value) internal {
+        ERC20Storage storage erc20 = erc20Storage();
+        ERC20BridgeableStorage storage bridge = bridgeStorage();
+
+        if (bridge.trustedBridges[msg.sender] == false) {
+            revert ERC20InvalidBridgeAccount(msg.sender);
+        }
+        if (_account == address(0)) {
+            revert ERC20InvalidReciever(address(0));
+        }
+        unchecked {
+            erc20.totalSupply += _value;
+            erc20.balanceOf[_account] += _value;
+        }
+
+        emit CrosschainMint(_account, _value, msg.sender);
+    }
+
+    /// @notice Internal crosschain burn logic. MUST be called only after validating caller.
+    /// @dev Decreases totalSupply and the `from` balance and emits CrosschainBurn.
+    /// @param _from The account to burn tokens from.
+    /// @param _value The amount to burn.
+    function crosschainBurn(address _from, uint256 _value) internal {
+        ERC20Storage storage erc20 = erc20Storage();
+        ERC20BridgeableStorage storage bridge = bridgeStorage();
+
+        if (bridge.trustedBridges[msg.sender] == false) {
+            revert ERC20InvalidBridgeAccount(msg.sender);
+        }
+        if (_from == address(0)) {
+            revert ERC20InvalidReciever(address(0));
+        }
+        uint256 accountBalance = erc20.balanceOf[_from];
+
+        if (accountBalance < _value) {
+            revert ERC20InsufficientBalance(_from, accountBalance, _value);
+        }
+        unchecked {
+            erc20.totalSupply -= _value;
+            erc20.balanceOf[_from] -= _value;
+        }
+
```

Comment:
To be compatible with the OpenZeppelin implementation we should also emit the Transfer event when burning the token.